### PR TITLE
User authentication on BE and FE

### PR DESCRIPTION
**Pull request by achromik**

### CODE REVIEW

File: backend/src/users/users.controller.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import { Controller } from '../interfaces/controller.interface';
+import userModel from './user.model';
+import { UserNotFoundException } from '../exceptions/UserExceptions';
+import { JWTExpiredException } from '../exceptions/JWTExceptions';
+
+export class UsersController implements Controller {
+    public path = '/users';
+    public router = express.Router();
+    private secretKey = process.env.SECRET_KEY || 'secret';
+    private user = userModel;
```
Comment (MarcinMiler) [inline backend/src/users/users.controller.ts]: You should use dependency injection to inject the user repository

File: backend/src/users/users.controller.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    ) => {
+        const today = new Date();
+
+        const userData = {
+            firstName: req.body.firstName,
+            lastName: req.body.lastName,
+            email: req.body.email,
+            password: req.body.password,
+            created: today,
+        };
```
Comment (Mazur-Sebastian) [inline backend/src/users/users.controller.ts]: You can add the user schema parameter timestamp, and the createdAt and updatedAt data will be created automatically in the Mongo document.

`export const UserSchema = new mongoose.Schema(
    {
       ...
    },
    {
        timestamps: true,
    },
);`

File: backend/src/users/users.controller.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    private registerUser = (
+        req: express.Request,
+        res: express.Response,
+        next: express.NextFunction,
+    ) => {
+        const today = new Date();
+
+        const userData = {
+            firstName: req.body.firstName,
```
Comment (Mazur-Sebastian) [inline backend/src/users/users.controller.ts]: I think body should by destructured

File: backend/src/users/users.controller.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            created: today,
+        };
+
+        this.user
+            .findOne({
+                email: req.body.email,
+            })
+            .then(user => {
+                if (!user) {
+                    bcrypt.hash(req.body.password, 10, (err, hash) => {
```
Comment (PiotrAdamus) [inline backend/src/users/users.controller.ts]: I think bcrypt can return promise, than uou can chain it with previous promises :)

File: backend/src/users/users.controller.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+import * as jwt from 'jsonwebtoken';
+import * as bcrypt from 'bcrypt';
+
+import { Controller } from '../interfaces/controller.interface';
+import userModel from './user.model';
+import { UserNotFoundException } from '../exceptions/UserExceptions';
+import { JWTExpiredException } from '../exceptions/JWTExceptions';
+
+export class UsersController implements Controller {
+    public path = '/users';
```
Comment (PiotrAdamus) [inline backend/src/users/users.controller.ts]: Mby use readonly here ?

**Pull request merged**